### DISCUSSION TITLE
command: webbutton<x>

### DISCUSSION THREAD

**Discussion by Mark-Muc** [arendst/Tasmota#19324] [Ideas]
Originally posted by Mark-Muc August 7, 2023
Hello .. just made a W32 board with port expander .. working fine and stable.
I've added 25 relays and 12 physical buttons .. so far so good..
I'am using the command: webbutton<x> to display some information/status on the button ..
now i found out .. it only works until <= 16 ..

webbutton17 Hallo!!
in the console. Result:

08:14:21.091 CMD: webbutton17 Hallo!!
08:14:21.099 MQT: stat/LCN_M5_12_Port/RESULT = {"Command":"Error"}
x untill 16 is working!!
Due to that u can choose Relays untill 28 .. so the command webbutton should be enhanced!!
Thx and Regards
Mark_Muc

Any News on this??

**Comment by Mark-Muc** [comment]
just asking .. any news on that??
Regards 
Mark_Muc

**Reply by sfromis** [reply]
No news. The max is indeed 16, and due to how the text pool is organized, it is not simply a question of changing the constant with the max number. Also, that text pool is of a limited total size, meaning that increasing number of stored strings would significantly increase the risk of exhausting what's available.

And as usual ... the rudimentary Tasmota web UI is not meant as a "device dashboard" for daily operation.

**Reply by barbudor** [reply]
I can have a look next week.
It would be easy to enhance that on ESP32 based on filesystem to store the extended text without increasing pressure on the Settings area

**Reply by Mark-Muc** [reply]
Thank u barbudor for Looking into it .. it will be on a ESP 32bit device , as mentioned!!
Mark_muc

**Reply by barbudor** [reply]
@Mark-Muc This has been merged and is now in the latest Development version
Until I update the docs, please read the details here : https://github.com/arendst/Tasmota/pull/19580

**Reply by Mark-Muc** [reply]
Thx barbudor!!
Just built a version .. and IT WORKS!!
![image](https://github.com/arendst/Tasmota/assets/67410847/127c5c3b-e051-4334-bde8-ec1016840b82)


Super and thank u!!
Mark_Muc
